src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -482,98 +461,92 @@ public override async Task RunAsync()
-            var isInProc8 = string.Equals(HostRuntime, DotnetConstants.InProc8HostRuntime, StringComparison.OrdinalIgnoreCase);
+            validateDotNet8ProjectEnablement ??= () => IsInProcDotNet8Enabled();
 
-            // If we are running an .NET isolated app and the user specifies inproc6 or inproc8, throw an error
-            if (isOutOfProc && (isInProc8 || isInProc6))
-            {
-                throw new CliException($"The runtime host value passed in, {HostRuntime}, is not a valid host version for your project. The host runtime is only valid for the worker runtime {WorkerRuntime.dotnet}");
-            }
-            // If we are not running a .NET app and the user specifies inproc6 or inproc8, throw an error
-            if (!isOutOfProc && !isInProc && (isInProc8 || isInProc6))
-            {
-                throw new CliException($"The runtime host value passed in, {HostRuntime}, is not a valid host version for your project. The runtime is only valid for {WorkerRuntime.dotnetIsolated} and {WorkerRuntime.dotnet}");
-            }
+            void ThrowCliException(string suffix) => throw new CliException($"The runtime argument value provided, '{HostRuntime}', is not valid for your project. {suffix}");
 
-            if (string.Equals(HostRuntime, "default", StringComparison.OrdinalIgnoreCase))
-            {
-                if (isInProc)
-                {
-                    throw new CliException($"The runtime host value passed in, default, is not a valid host version for your project. For the default host runtime, the worker runtime must be set to {WorkerRuntime.dotnetIsolated}.");
-                }
-                PrintVerboseForHostSelection(isVerbose, "out-of-process");
-                return;
-            }
-            else if (isInProc8)
+            if (DotnetConstants.ValidRuntimeValues.Contains(HostRuntime, StringComparer.OrdinalIgnoreCase) == false)
             {
-                if (!await IsInProcNet8Enabled())
-                {
-                    throw new CliException($"The runtime host value passed in, {DotnetConstants.InProc8HostRuntime}, is not a valid host version for your project. For the {DotnetConstants.InProc8HostRuntime} runtime, the {Constants.FunctionsInProcNet8Enabled} variable must be set while running a .NET 8 in-proc app.");
-                }
-                PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc8HostRuntime);
-                return;
+                ThrowCliException($"The runtime argument value provided, '{HostRuntime}', is not valid. Valid values are '{string.Join("', '", DotnetConstants.ValidRuntimeValues)}'.");
             }
-            else if (isInProc6)
+
+            var isInproc6Runtime = string.Equals(HostRuntime, DotnetConstants.InProc6HostRuntime, StringComparison.OrdinalIgnoreCase);
+            var isInproc8Runtime = string.Equals(HostRuntime, DotnetConstants.InProc8HostRuntime, StringComparison.OrdinalIgnoreCase);
```

Comment:
worth renaming these to `userSpecifiedInproc6Runtime` or something? "Is" feels ambiguous here.